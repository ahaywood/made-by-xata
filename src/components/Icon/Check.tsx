interface CheckProps {
  height?: number;
  width?: number;
}

const Check = ({ height = 24, width = 24 }: CheckProps) => (
  <svg
    width={width}
    height={height}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    data-testid="checkIcon"
  >
    <title>Check</title>
    <path d="M9.00063 16.17L5.53063 12.7C5.14063 12.31 4.51063 12.31 4.12062 12.7C3.73063 13.09 3.73063 13.72 4.12062 14.11L8.30063 18.29C8.69063 18.68 9.32063 18.68 9.71063 18.29L20.2906 7.71001C20.6806 7.32001 20.6806 6.69001 20.2906 6.30001C19.9006 5.91001 19.2706 5.91001 18.8806 6.30001L9.00063 16.17Z" />
  </svg>
);

export { Check };
