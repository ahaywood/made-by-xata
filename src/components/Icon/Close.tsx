interface CloseProps {
  height?: number;
  width?: number;
}

const Close = ({ height = 24, width = 24 }: CloseProps) => {
  return (
    <svg width={width} height={height} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" data-testid="closeIcon">
      <title>Close</title>
      <path d="M24.4009 7.61336C23.8809 7.09336 23.0409 7.09336 22.5209 7.61336L16.0009 14.12L9.48094 7.60002C8.96094 7.08002 8.12094 7.08002 7.60094 7.60002C7.08094 8.12002 7.08094 8.96002 7.60094 9.48002L14.1209 16L7.60094 22.52C7.08094 23.04 7.08094 23.88 7.60094 24.4C8.12094 24.92 8.96094 24.92 9.48094 24.4L16.0009 17.88L22.5209 24.4C23.0409 24.92 23.8809 24.92 24.4009 24.4C24.9209 23.88 24.9209 23.04 24.4009 22.52L17.8809 16L24.4009 9.48002C24.9076 8.97336 24.9076 8.12002 24.4009 7.61336Z" />
    </svg>
  )
}

export {Close}