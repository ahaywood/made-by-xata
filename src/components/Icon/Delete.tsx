interface DeleteProps {
  height?: number;
  width?: number;
}

const Delete = ({ height = 24, width = 24 }: DeleteProps) => {
  return (
    <svg width={width} height={height} viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" data-testid="deleteIcon">
      <title>Delete</title>
      <path d="M10 23C10 24.1 10.9 25 12 25H20C21.1 25 22 24.1 22 23V13C22 11.9 21.1 11 20 11H12C10.9 11 10 11.9 10 13V23ZM13 13H19C19.55 13 20 13.45 20 14V22C20 22.55 19.55 23 19 23H13C12.45 23 12 22.55 12 22V14C12 13.45 12.45 13 13 13ZM19.5 8L18.79 7.29C18.61 7.11 18.35 7 18.09 7H13.91C13.65 7 13.39 7.11 13.21 7.29L12.5 8H10C9.45 8 9 8.45 9 9C9 9.55 9.45 10 10 10H22C22.55 10 23 9.55 23 9C23 8.45 22.55 8 22 8H19.5Z" />
    </svg>
  )
}

export {Delete}