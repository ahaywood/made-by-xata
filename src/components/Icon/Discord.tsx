interface DiscordProps {
  height?: number;
  width?: number;
}

const Discord = ({ height = 24, width = 24 }: DiscordProps) => (
  <svg
    width={width}
    height={height}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    data-testid="discordIcon"
  >
    <title>Discord</title>
    <path d="M18.3042 6.50484C17.1447 5.97285 15.9014 5.5809 14.6015 5.35641C14.5778 5.35208 14.5541 5.36291 14.5419 5.38456C14.382 5.66896 14.2049 6.03997 14.0809 6.33159C12.6827 6.12227 11.2917 6.12227 9.92217 6.33159C9.79811 6.03349 9.61457 5.66896 9.45395 5.38456C9.44175 5.36363 9.4181 5.3528 9.39443 5.35641C8.09519 5.58018 6.85188 5.97213 5.69173 6.50484C5.68169 6.50917 5.67308 6.5164 5.66737 6.52577C3.30907 10.049 2.66304 13.4857 2.97996 16.8797C2.98139 16.8963 2.99071 16.9122 3.00362 16.9223C4.55956 18.0649 6.06675 18.7586 7.54595 19.2184C7.56963 19.2256 7.59471 19.217 7.60977 19.1975C7.95968 18.7197 8.27159 18.2158 8.53902 17.686C8.5548 17.6549 8.53974 17.6181 8.50748 17.6059C8.01274 17.4182 7.54165 17.1894 7.08849 16.9295C7.05265 16.9086 7.04978 16.8573 7.08275 16.8328C7.17811 16.7613 7.2735 16.687 7.36456 16.6119C7.38103 16.5982 7.40399 16.5953 7.42336 16.6039C10.4004 17.9632 13.6234 17.9632 16.5654 16.6039C16.5847 16.5946 16.6077 16.5975 16.6249 16.6112C16.716 16.6862 16.8113 16.7613 16.9074 16.8328C16.9404 16.8573 16.9382 16.9086 16.9024 16.9295C16.4492 17.1944 15.9781 17.4182 15.4827 17.6051C15.4504 17.6174 15.4361 17.6549 15.4518 17.686C15.725 18.2151 16.0369 18.7189 16.3804 19.1968C16.3947 19.217 16.4205 19.2256 16.4442 19.2184C17.9306 18.7586 19.4378 18.0649 20.9937 16.9223C21.0073 16.9122 21.0159 16.897 21.0174 16.8804C21.3967 12.9565 20.3821 9.54808 18.3278 6.52649C18.3228 6.5164 18.3142 6.50917 18.3042 6.50484ZM8.9836 14.8131C8.0873 14.8131 7.34877 13.9902 7.34877 12.9797C7.34877 11.9691 8.07298 11.1462 8.9836 11.1462C9.90137 11.1462 10.6327 11.9763 10.6184 12.9797C10.6184 13.9902 9.89419 14.8131 8.9836 14.8131ZM15.0281 14.8131C14.1318 14.8131 13.3933 13.9902 13.3933 12.9797C13.3933 11.9691 14.1175 11.1462 15.0281 11.1462C15.9459 11.1462 16.6772 11.9763 16.6629 12.9797C16.6629 13.9902 15.9459 14.8131 15.0281 14.8131Z" />
  </svg>
);

export { Discord };
