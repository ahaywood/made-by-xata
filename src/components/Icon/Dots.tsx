interface DotsProps {
  height?: number;
  width?: number;
}

const Dots = ({ height = 24, width = 24 }: DotsProps) => (
  <svg
    width={width}
    height={height}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    data-testid="dotsIcon"
  >
    <title>Dots</title>
    <circle cx="4.5" cy="12" r="2.25" />
    <circle cx="12" cy="12" r="2.25" />
    <circle cx="19.5" cy="12" r="2.25" />
  </svg>
);

export { Dots };
