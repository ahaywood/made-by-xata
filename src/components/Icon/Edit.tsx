interface EditProps {
  height?: number;
  width?: number;
}

const Edit = ({ height = 24, width = 24 }: EditProps) => (
  <svg
    width={width}
    height={height}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    data-testid="editIcon"
  >
    <title>Edit</title>
    <path d="M5.25 16.095V18.375C5.25 18.585 5.415 18.75 5.625 18.75H7.905C8.0025 18.75 8.1 18.7125 8.1675 18.6375L16.3575 10.455L13.545 7.64248L5.3625 15.825C5.2875 15.9 5.25 15.99 5.25 16.095ZM18.5325 8.27998C18.825 7.98748 18.825 7.51498 18.5325 7.22248L16.7775 5.46748C16.485 5.17498 16.0125 5.17498 15.72 5.46748L14.3475 6.83998L17.16 9.65248L18.5325 8.27998Z" />
  </svg>
);

export { Edit };
