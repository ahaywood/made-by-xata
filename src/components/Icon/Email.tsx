interface EmailProps {
  height?: number;
  width?: number;
}

const Email = ({ height = 24, width = 24 }: EmailProps) => (
  <svg
    width={width}
    height={height}
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    data-testid="emailIcon"
  >
    <title>Email</title>
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M6 6.75C5.58921 6.75 5.25 7.08921 5.25 7.5V16.5C5.25 16.9108 5.58921 17.25 6 17.25H18C18.4108 17.25 18.75 16.9108 18.75 16.5V7.5C18.75 7.08921 18.4108 6.75 18 6.75H6ZM3.75 7.5C3.75 6.26079 4.76079 5.25 6 5.25H18C19.2392 5.25 20.25 6.26079 20.25 7.5V16.5C20.25 17.7392 19.2392 18.75 18 18.75H6C4.76079 18.75 3.75 17.7392 3.75 16.5V7.5Z"
    />
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M3.88564 7.06985C4.12318 6.73051 4.59082 6.64798 4.93016 6.88552L12.0001 11.8345L19.07 6.88552C19.4093 6.64798 19.877 6.73051 20.1145 7.06985C20.352 7.40918 20.2695 7.87683 19.9302 8.11437L12.4302 13.3644C12.1719 13.5451 11.8282 13.5451 11.57 13.3644L4.06997 8.11437C3.73063 7.87683 3.6481 7.40918 3.88564 7.06985Z"
    />
  </svg>
);

export { Email };
