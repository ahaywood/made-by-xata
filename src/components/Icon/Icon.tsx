import { Check } from "./Check";
import { Close } from "./Close";
import { Delete } from "./Delete";
import { Discord } from "./Discord";
import { Dots } from "./Dots";
import { Edit } from "./Edit";
import { Email } from "./Email";
import { GitHub } from "./GitHub";
import { LinkedIn } from "./Linkedin";
import { Twitter } from "./Twitter";
import { Upload } from "./Upload";

interface IconProps {
  height?: number;
  name: string;
  width?: number;
}

const Icon = ({ height = 24, name, width = 24 }: IconProps) => {
  switch (name.toLowerCase()) {
    case "check":
      return <Check height={height} width={width} />
    case "close":
      return <Close height={height} width={width} />
    case "delete":
      return <Delete height={height} width={width} />
    case "discord":
      return <Discord height={height} width={width} />
    case "dots":
      return <Dots height={height} width={width} />
    case "edit":
      return <Edit height={height} width={width} />
    case "email":
      return <Email height={height} width={width} />
    case "github":
      return <GitHub height={height} width={width} />
    case "linkedin":
      return <LinkedIn height={height} width={width} />
    case "twitter":
      return <Twitter height={height} width={width} />
    case "upload":
      return <Upload height={height} width={width} />
    default:
      return <div />
  }
}

export { Icon }