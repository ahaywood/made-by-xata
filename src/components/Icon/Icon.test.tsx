import { render, screen } from '@testing-library/react';
import { Icon } from './Icon';

describe('Icon', () => {
  it('renders correctly', () => {
    expect(() => {
      render(<Icon name="check" />);
    }).not.toThrow();
  });

  describe('Check Icon', () => {
    it('rendering the check icon', () => {
      render(<Icon name="check" />);
      expect(screen.getByTestId('checkIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="check" />);
      }).toMatchSnapshot();
    });
  });

  describe('Close Icon', () => {
    it('rendering the close icon', () => {
      render(<Icon name="close" />);
      expect(screen.getByTestId('closeIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="close" />);
      }).toMatchSnapshot();
    });
  });

  describe('Delete Icon', () => {
    it('rendering the delete icon', () => {
      render(<Icon name="delete" />);
      expect(screen.getByTestId('deleteIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="delete" />);
      }).toMatchSnapshot();
    });
  });

  describe('Discord Icon', () => {
    it('rendering the discord icon', () => {
      render(<Icon name="discord" />);
      expect(screen.getByTestId('discordIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="discord" />);
      }).toMatchSnapshot();
    });
  });

  describe('Dots Icon', () => {
    it('rendering the dots icon', () => {
      render(<Icon name="dots" />);
      expect(screen.getByTestId('dotsIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="dots" />);
      }).toMatchSnapshot();
    });
  });

  describe('Edit Icon', () => {
    it('rendering the edit icon', () => {
      render(<Icon name="edit" />);
      expect(screen.getByTestId('editIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="edit" />);
      }).toMatchSnapshot();
    });
  });

  describe('Email Icon', () => {
    it('rendering the email icon', () => {
      render(<Icon name="email" />);
      expect(screen.getByTestId('emailIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="email" />);
      }).toMatchSnapshot();
    });
  });

  describe('Filter Icon', () => {
    it('rendering the filter icon', () => {
      render(<Icon name="filter" />);
      expect(screen.getByTestId('filterIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="filter" />);
      }).toMatchSnapshot();
    });
  });

  describe('GitHub Icon', () => {
    it('rendering the github icon', () => {
      render(<Icon name="github" />);
      expect(screen.getByTestId('githubIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="github" />);
      }).toMatchSnapshot();
    });
  });

  describe('LinkedIn Icon', () => {
    it('rendering the linkedin icon', () => {
      render(<Icon name="linkedin" />);
      expect(screen.getByTestId('linkedinIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="linkedin" />);
      }).toMatchSnapshot();
    });
  });

  describe('Search Icon', () => {
    it('rendering the search icon', () => {
      render(<Icon name="search" />);
      expect(screen.getByTestId('searchIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="search" />);
      }).toMatchSnapshot();
    });
  });

  describe('Twitter Icon', () => {
    it('rendering the twitter icon', () => {
      render(<Icon name="twitter" />);
      expect(screen.getByTestId('twitterIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="twitter" />);
      }).toMatchSnapshot();
    });
  });

  describe('Upload Icon', () => {
    it('rendering the upload icon', () => {
      render(<Icon name="upload" />);
      expect(screen.getByTestId('uploadIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="upload" />);
      }).toMatchSnapshot();
    });
  });

  describe('Vertical Dots Icon', () => {
    it('rendering the vertical dots icon', () => {
      render(<Icon name="verticalDots" />);
      expect(screen.getByTestId('verticalDotsIcon')).toBeInTheDocument();
    });

    it('renders correctly', () => {
      expect(() => {
        render(<Icon name="verticalDots" />);
      }).toMatchSnapshot();
    });
  });
});
