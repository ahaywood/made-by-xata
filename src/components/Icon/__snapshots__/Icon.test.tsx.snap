// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Check Icon renders correctly 1`] = `[Function]`;

exports[`Icon Close Icon renders correctly 1`] = `[Function]`;

exports[`Icon Delete Icon renders correctly 1`] = `[Function]`;

exports[`Icon Discord Icon renders correctly 1`] = `[Function]`;

exports[`Icon Dots Icon renders correctly 1`] = `[Function]`;

exports[`Icon Edit Icon renders correctly 1`] = `[Function]`;

exports[`Icon Email Icon renders correctly 1`] = `[Function]`;

exports[`Icon Filter Icon renders correctly 1`] = `[Function]`;

exports[`Icon GitHub Icon renders correctly 1`] = `[Function]`;

exports[`Icon LinkedIn Icon renders correctly 1`] = `[Function]`;

exports[`Icon Search Icon renders correctly 1`] = `[Function]`;

exports[`Icon Twitter Icon renders correctly 1`] = `[Function]`;

exports[`Icon Upload Icon renders correctly 1`] = `[Function]`;

exports[`Icon Vertical Dots Icon renders correctly 1`] = `[Function]`;
